### PR TITLE
[auth][photos] Support for passkey

### PR DESCRIPTION
**Pull request by prateekmedia**
## Description

Passkey implementation (similar will be done in ente Photos)

## Type of Change

- [ ] 🖼️ New icon
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/l10n/arb/app_en.arb [hunk 1]
```diff
@@ -404,5 +404,7 @@
   "signOutOtherDevices": "Sign out other devices",
   "doNotSignOut": "Do not sign out",
   "hearUsWhereTitle": "How did you hear about Ente? (optional)",
-  "hearUsExplanation": "We don't track app installs. It'd help if you told us where you found us!"
-}
+  "hearUsExplanation": "We don't track app installs. It'd help if you told us where you found us!",
+  "waitingForBrowserRequest": "Waiting for browser request...",
+  "launchPasskeyUrlAgain": "Launch passkey url again"
```
Comment (dannya) [inline lib/l10n/arb/app_en.arb]: Would be better if `URL` is capitalized in the message string.

File: lib/services/user_service.dart [hunk 2]
```diff
@@ -264,6 +265,40 @@ class UserService {
...
+    BuildContext context,
+    Map response,
+    String userPassword,
+    Uint8List keyEncryptionKey,
+  ) async {
+    await _saveConfiguration(response);
+
+    Widget page;
+    if (Configuration.instance.getEncryptedToken() != null) {
+      // print(userPassword);
```
Comment (dannya) [inline lib/services/user_service.dart]: Probably best to remove these 4 commented out lines?

File: lib/l10n/arb/app_en.arb (hunk 1 above)
Comment (dannya) [reply lib/l10n/arb/app_en.arb]: @prateekmedia that was quick :)

File: lib/models/key_attributes.dart [hunk 3]
```diff
@@ -99,4 +99,9 @@ class KeyAttributes {
       recoveryKeyDecryptionNonce ?? this.recoveryKeyDecryptionNonce,
     );
   }
+
+  @override
+  String toString() {
+    return 'KeyAttributes(kekSalt: $kekSalt, encryptedKey: $encryptedKey, keyDecryptionNonce: $keyDecryptionNonce, publicKey: $publicKey, encryptedSecretKey: $encryptedSecretKey, secretKeyDecryptionNonce: $secretKeyDecryptionNonce, memLimit: $memLimit, opsLimit: $opsLimit, masterKeyEncryptedWithRecoveryKey: $masterKeyEncryptedWithRecoveryKey, masterKeyDecryptionNonce: $masterKeyDecryptionNonce, recoveryKeyEncryptedWithMasterKey: $recoveryKeyEncryptedWithMasterKey, recoveryKeyDecryptionNonce: $recoveryKeyDecryptionNonce)';
+  }
```
Comment (ua741) [inline lib/models/key_attributes.dart]: undo

**Review by ua741 — APPROVED** [review]

**Pull request merged**